### PR TITLE
[PrintAsObjC] Use 'unsafe_unretained' to print 'unowned', not 'assign'

### PR DESCRIPTION
**Pull request by jrose-apple**
The ObjC generator previously preserved a subtle difference between `unowned` and `unowned(unsafe)` / Unmanaged by printing the former as `assign` and the latter as `unsafe_unretained`. Upstream Clang, however, has gotten a new warning to discourage the use of `assign` with reference-countable types at all. Since it was always a subtle distinction, just go with the new convention and print `unsafe_unretained` for `unowned` properties as well.

rdar://problem/44290715

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test Linux

**Comment by jrose-apple** [comment]
@swift-ci Please test macOS

File: lib/PrintAsObjC/PrintAsObjC.cpp [hunk 1]
```diff
@@ -1077,9 +1074,11 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
...
         os << ", weak";
       }
-    } else if (ty->is<UnownedStorageType>()) {
-      os << ", assign";
-    } else if (ty->is<UnmanagedStorageType>()) {
+    } else if (ty->is<UnownedStorageType>()|| ty->is<UnmanagedStorageType>()) {
+      // We treat "unowned" as "unsafe_unretained" (even though it's more like
+      // "safe_unretained") because we want people to think twice about
+      // allowing that object to disappear. "unowned(unsafe)" (and Unmanaged,
+      // handled below) really are "unsafe_unretained".
```
Comment (rjmccall) [inline lib/PrintAsObjC/PrintAsObjC.cpp]: Can this reasonably be turned into an exhaustive switch over reference storage kinds?

**Review by rjmccall — APPROVED** [review]
Comment aside, LGTM.  I filed rdar://44290715 before I caught up on my code-review inbox and saw that you'd done work on this, though, sorry.

**Comment by jrose-apple** [comment]
Good suggestion, spotted another reasonable cleanup while I was doing that.

@swift-ci Please smoke test

**Pull request merged**